apps/managedidentity/managedidentity.go

```diff
@@ -35,23 +38,64 @@ const (
 	CloudShell    Source = "CloudShell"
 	AppService    Source = "AppService"
 
-	// General request querry parameter names
-	metaHTTPHeaderName            = "Metadata"
-	apiVersionQuerryParameterName = "api-version"
-	resourceQuerryParameterName   = "resource"
+	// General request query parameter names
+	metaHTTPHeaderName           = "Metadata"
+	apiVersionQueryParameterName = "api-version"
+	resourceQueryParameterName   = "resource"
+	wwwAuthenticateHeaderName    = "www-authenticate"
 
-	// UAMI querry parameter name
+	// UAMI query parameter name
 	miQueryParameterClientId   = "client_id"
 	miQueryParameterObjectId   = "object_id"
 	miQueryParameterResourceId = "msi_res_id"
 
 	// IMDS
-	imdsEndpoint   = "http://169.254.169.254/metadata/identity/oauth2/token"
-	imdsAPIVersion = "2018-02-01"
-
+	imdsDefaultEndpoint           = "http://169.254.169.254/metadata/identity/oauth2/token"
+	imdsAPIVersion                = "2018-02-01"
 	systemAssignedManagedIdentity = "system_assigned_managed_identity"
+
+	// Azure Arc
+	azureArcEndpoint               = "http://127.0.0.1:40342/metadata/identity/oauth2/token"
+	azureArcAPIVersion             = "2020-06-01"
+	azureArcFileExtension          = ".key"
+	azureArcMaxFileSizeBytes int64 = 4096
+	linuxTokenPath                 = "/var/opt/azcmagent/tokens"
+	linuxHimdsPath                 = "/opt/azcmagent/bin/himds"
+	azureConnectedMachine          = "AzureConnectedMachine"
```

Comment:
```suggestion
	azureConnectedMachine          = "AzureConnectedMachineAgent"
```

Reply:
Have you tested the flow manually? If not I would recommend to test it before merging.